plugins/woocommerce/client/blocks/assets/js/shared/redux-stores/canvas-mode/index.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { createReduxStore, register, select, dispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ACTION_SET_CANVAS_MODE, STORE_NAME } from './constants';
+import type { CanvasModeState } from './types';
+
+type StoreState = CanvasModeState;
+
+type Actions = {
+	type: typeof ACTION_SET_CANVAS_MODE;
+	isEditMode: boolean;
+};
+
+const DEFAULT_STATE: StoreState = {
+	isEditMode: false,
+};
+
+const actions = {
+	setCanvasMode( isEditMode: boolean ) {
+		return {
+			type: ACTION_SET_CANVAS_MODE,
+			isEditMode,
+		};
+	},
+};
+
+const selectors = {
+	isEditMode() {
+		// Always check the current URL for the most up-to-date state.
+		try {
+			const url = new URL( window.location.href );
+			const canvas = url.searchParams.get( 'canvas' );
+			return canvas === 'edit';
+		} catch ( error ) {
+			// If URL parsing fails, return default state.
+			return false;
+		}
+	},
+};
```

Comment:
You should get the value from the state:

```suggestion
const selectors = {
	isEditMode( state: StoreState ) {
		return state.isEditMode;
	},
};
```